File: pmd/java/listas/listaex04/ex1/Produto.java
package listas.listaex04.ex1;

public class Produto<T> implements Comparable<Produto<T>> {

  private T codigo;
  private String nome;
  private float valor;

  public Produto(T codigo, String nome, float valor) {
    this.codigo = codigo;
    this.nome = nome;
    this.valor = valor;
  }

  public String toString() {
    return "Produto 	{ codigo = '" + codigo + "', nome = '"
        + nome + "', valor = '" + valor + "'}";
  }

  public int compareTo(Produto<T> p) {
    int r = this.nome.compareTo(p.nome);
    return r != 0 ? r : (int) (this.valor - p.valor);
  }

}

File: pmd/python/exercicios/listaex04/ex1/produto.py
class Produto:

  def __init__(self, codigo = 0, nome = "", valor = 0.0):
    self.codigo = codigo
    self.nome = nome
    self.valor = valor
  
  def __str__(self):
    return "Produto { codigo = '%d', nome = '%s', valor = '%.2f'}" % \
      (self.codigo, self.nome, self.valor)

  def __lt__(self, other):
    r = self.nome < other.nome
    return  r if r else self.valor < other.valor

  def __le__(self, other):
    r = self.nome <= other.nome
    return  r if r else self.valor <= other.valor

  def __gt__(self, other):
    r = self.nome > other.nome
    return  r if r else self.valor > other.valor

  def __ge__(self, other):
    r = self.nome >= other.nome
    return  r if r else self.valor >= other.valor

  def __eq__(self, other):
    r = self.nome == other.nome
    return  r if r else self.valor == other.valor

  def __ne__(self, other):
    r = self.nome != other.nome
    return  r if r else self.valor != other.valor
